nexus/db-queries/src/db/datastore/identity_provider.rs

```diff
@@ -11,18 +11,66 @@ use crate::db;
 use crate::db::error::public_error_from_diesel;
 use crate::db::error::ErrorHandler;
 use crate::db::identity::Resource;
+use crate::db::lookup::LookupPath;
+use crate::db::model;
 use crate::db::model::IdentityProvider;
 use crate::db::model::Name;
 use crate::db::pagination::paginated;
 use async_bb8_diesel::AsyncRunQueryDsl;
 use diesel::prelude::*;
+use nexus_auth::authn::silos::IdentityProviderType;
 use omicron_common::api::external::http_pagination::PaginatedBy;
 use omicron_common::api::external::CreateResult;
 use omicron_common::api::external::ListResultVec;
+use omicron_common::api::external::LookupResult;
 use omicron_common::api::external::ResourceType;
 use ref_cast::RefCast;
 
 impl DataStore {
+    pub async fn identity_provider_lookup(
```

Comment:
This method was moved exactly from `IdentityProvider::lookup` to here (and `DataStore` is now a `&self` parameter instead of the first argument)